### DISCUSSION TITLE
Window上使用Android Studio编译后，除了默认的英文输入，其他所有的输入法添加后都是(not available)

### DISCUSSION THREAD

**Discussion by QiXuanZhang-zqx** [fcitx5-android/fcitx5-android#573] [Q&A | 问答] [answered]
APK的release版本大小只有8Mb，像是其他输入法的配置都没有载入成功，但是已经按照项目流程进行配置了。请问如何解决

**Comment by rocka** [comment]
应该是漏了一步 On Windows, you may need to regenerate symlinks to submodules

README.md 里面有

https://github.com/fcitx5-android/fcitx5-android/blob/d07838d86878717190709c49dce5fa143846a33e/README.md?plain=1#L99-L120

https://github.com/fcitx5-android/fcitx5-android/blob/d07838d86878717190709c49dce5fa143846a33e/.github/workflows/pull_request.yml#L29-L33

**Reply by QiXuanZhang-zqx** [reply]
这个流程执行过了，但是发现还是有这样的问题存在。Windows中编译项目，提示一些文件相对路径找不到我把这里修改到编译不报错，但是编译出的apk，无法使用其他的输入法，是否与这个有关系。

**Reply by QiXuanZhang-zqx** [reply]
想问问，项目中是否还有一些配置文件可以检查，感觉还是有些地方没有配置上

**Reply by rocka** [reply]
> 提示一些文件相对路径找不到我把这里修改到编译不报错

哪里报错，报错内容是什么，你怎么改的

**Reply by QiXuanZhang-zqx** [reply]
主要有以下的修改
1、Fcitx5Macros.cmake和Fcitx5ComplierSettings.cmake，本身内部包含的相对路径，但是不懂为啥始终找不到位置，我修改为了“include(${CMAKE_CURRENT_LIST_DIR}/../../../cpp/fcitx5/cmake/Fcitx5CompilerSettings.cmake)"后不报错
2、在fcitx-chinese-addons-src-main-cpp的CMakeList中，boost库的版本和路径如果不设置，编译的cmake也会报错，我将版本修改为导入项目时boost1.83.0，还有在这个cmake中为Boost_INCLUDE_DIRS设置了我本地的绝对路径，cmake不报错
3、在2的基础上，例如pinyin中的CMakeLists.txt报错，提示找不到boost库的文件（修改2也是解决这个问题），我在这个CMakeLists中增加了”（target_include_directories(pinyin PUBLIC ${Boost_INCLUDE_DIRS})“，其他文件夹中的有相同报错的，也使用该方法，最后编译能通过，可以产生APK文件，但是就有标题这样的问题了，目前还发现pinyin的配置界面也是处于无法加载的状态，似乎还是某个地方没有配置好。

**Reply by QiXuanZhang-zqx** [reply]
assets中似乎已经将lib中的conf文件都准备好了，像是在最后生成apk时，这些文件没有被打包进去

**Reply by QiXuanZhang-zqx** [reply]
估计还是Windows平台上软链接的问题，assets中资源并不完整，我将你发布的realease版本的APK中的assets文件进行替换后，本地编译出的APK正常了。感谢你的帮助🌹

**Reply by QiXuanZhang-zqx** [reply]
大佬，还有一个问题，我想在apk安装时，默认添加pinyin请问该查看哪部分的代码。对kotlin不是很熟悉

**Reply by rocka** [reply, marked as answer]
应该还是没复制全。默认已经有根据系统语言自动添加对应的输入法的功能了，比如简体中文 `zh_CN` 就自动添加拼音

https://github.com/fcitx/fcitx5/blob/5.1.10/data/default/zh_CN#L2

在 apk 中是 assets/usr/share/fcitx5/default 这个目录

**Reply by QiXuanZhang-zqx** [reply]
是的，确实是这个defaulet文件没有利用好。感谢~
